### PR TITLE
Fix various "incompatible" messages for plugin manager

### PR DESCRIPTION
**Pull request by daniel-beck**
This is essentially two changes combined:

* Broaden the meaning of compatibility warnings (`compatibleSince`) to match how it's sometimes used – to warn about an incompatible behavior change.
* Replace the term "dependent" (AFAICT, that which _has_ a dependency, rather than _is_ a dependency) with "dependencies"

Plus made some similar messages more consistent in wording.

### Proposed changelog entries

Too minor

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [n/a] JIRA issue is well described
- [n/a] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [ ] Appropriate autotests or explanation to why this change has no tests
- [n/a] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a JIRA issue should exist and be labeled as `lts-candidate`

### CODE REVIEW

**Review by MarkEWaite — APPROVED** [review]
I like the improved phrasing.

**Review by res0nance — APPROVED** [review]

**Review by varyvol — APPROVED** [review]

**Review by timja — APPROVED** [review]

File: core/src/main/resources/hudson/PluginManager/table.properties
Comment (oleg-nenashev) [inline core/src/main/resources/hudson/PluginManager/table.properties]:
```suggestion
 Warning: this plugin has dependencies on other plugins that require Java {0} or newer. \
```

I am not a native speaker tho :)

**Comment by fqueiruga** [comment]
Is there anything that blocks this PR? Seems a rather straightforward one with many approvals.

File: core/src/main/resources/hudson/PluginManager/table.properties
Comment (timja) [reply core/src/main/resources/hudson/PluginManager/table.properties:45]: I think this wording is better the other does work but doesn't read as well.

**Comment by timja** [comment]
@daniel-beck do you want to integrate the suggested wording?

**Comment by daniel-beck** [comment]
Done.

**Review by timja — APPROVED** [review]

**Comment by timja** [comment]
Thanks for the PR! Let's merge this tomorrow if there's no negative feedback

**Pull request merged**
